File: C#/Practice/LeetCode/Easy/Add Digits.cs
// https://leetcode.com/problems/add-digits/description/

public class Solution {
    public int AddDigits(int num) {
        string numStr = num.ToString();
        char[] digits = numStr.ToCharArray();
        int sum = 0;

        while (digits.Length > 1) {
            for (int i = 0; i < digits.Length; i++) {
                sum += int.Parse(digits[i].ToString());
            }

            numStr = sum.ToString();
            digits = numStr.ToCharArray();
            sum = 0;
        }

        return int.Parse(digits[0].ToString());
    }
}

File: TypeScript/Practice/LeetCode/Easy/Add Digits.ts
// https://leetcode.com/problems/add-digits/description/

function addDigits(num: number): number {
    let digits: string[] = new String(num).split("");
    let sum: number = 0;

    while (digits.length > 1) {
        for (let i = 0; i < digits.length; i++) {
            sum += Number(digits[i])
        }

        digits = new String(sum).split("");
        sum = 0;
    }

    return Number(digits[0]);
};